backend/flask_routes/app.py

```diff
@@ -680,9 +681,103 @@ def filter_existing_goals(create_goal_request):
+the past two most recent transaction syncs with the context of the goal amount
+"""
+def format_updates(userId):
+    update_msg = ''
+    user = User.query.get(userId)
+    goals = user.goals
+    new_transaction = Transactions.query.filter_by(userId=user.id).order_by(Transactions.time.desc()).first()
+    # this gets last weeks transaction (the one before newest / second row) by simply skipping first row and getting the next row
+    last_transaction = Transactions.query.filter_by(userId=user.id).order_by(Transactions.time.desc()).offset(1).limit(1).first()
+    for goal in goals:
+        category = goal.category
+        update_msg += 'category: ' + category + ' targeted change in expenditure: ' + str(goal.target) +  ' this weeks percent: '+  str(getattr(new_transaction, category)) + ' last week\'s percent: ' + str(getattr(last_transaction, category)) + '\n'
+    return update_msg
+
+
+"""
+this function handles the transaction sync endpoint for updating user transactions
+this is designed as a helper so it can be used both by the flask route and in schedular cron job
+"""
+def transactions_sync(userId):
+    try:
+        user = User.query.get(userId)
+        # decrypt the access token from the database to use it
+        encrypted_token = user.token
+        key = os.getenv('ENCRYPTION_KEY')
+        cipher = Fernet(key)
+        decrypted_token = cipher.decrypt(encrypted_token.encode('utf-8')).decode('utf-8')
+
+        request = TransactionsSyncRequest(access_token=decrypted_token)
+        transactions = plaid_client.transactions_sync(request)
+        # transaction object is by default non-serialable hence why we transform it to a dictionary
+        transactions_data = {"transactions": []}
+        for transaction in transactions.added:
+            transactions_data["transactions"].append(transaction.to_dict())
+        clean_data = clean_transaction_data(transactions_data)
+        user_transaction_data = aggregate_user_data(clean_data)
+        db_status = save_transaction(userId, user_transaction_data)
+        return jsonify({'message': db_status, 'data': user_transaction_data})
+    except Exception as e:
+        print(f"error at transactions_sync: {str(e)}")
+        return jsonify({'error': 'error at transactions_sync'}), 500
+
+
+"""
+this helper function is called once per week to trigger an email update message for 
+every user that updates them on progress towards the goals they're tracking
+"""
+def weekly_sync():
+   with app.app_context():
+        try:
+            users = User.query.all()
+            for user in users:
+                id = user.id
+                transactions_sync(id)
+                send_goal_update_msg(id)
+        except Exception as e:
+            return jsonify({'error': 'error at weekly_sync'}), 500
+
+
+# [weekly_sync] is called every sunday at 11:59PM PDT to notify users of goal progress and updates
+scheduler.add_job(
```

Comment:
@carsonwolber This won't actually run every Sunday unless you deploy it to an always-on server, right? Just want to clarify

Reply:
I'm pretty sure this would run every Sunday as long as my server is running at the time it's set to run, if the server isn't running on my machine at 11:59 on Sunday then yeah there's no backups to keep the server live.